features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/DatadogFlagsClient.kt

```diff
@@ -0,0 +1,153 @@
+     */
+    override fun resolveStringValue(flagKey: String, defaultValue: String): String {
+        val precomputedData = flagsRepository.getPrecomputedFlag(flagKey)
+        return precomputedData?.variationValue ?: defaultValue
+    }
+
+    /**
+     * Resolves an integer flag value from the local repository.
+     *
+     * This method attempts to parse the stored string value as an integer.
+     * If parsing fails, the default value is returned. This method is thread-safe
+     * and performs no network operations.
+     *
+     * @param flagKey The name of the flag to query. Cannot be null.
+     * @param defaultValue The value to return if the flag cannot be retrieved or parsed.
+     * @return The integer value of the flag, or the default value if unavailable or unparseable.
+     */
+    override fun resolveIntValue(flagKey: String, defaultValue: Int): Int {
+        val precomputedData = flagsRepository.getPrecomputedFlag(flagKey)
+        return precomputedData?.variationValue?.toIntOrNull() ?: defaultValue
+    }
+
+    /**
+     * Resolves a numeric flag value from the local repository.
+     *
+     * This method attempts to parse the stored string value as a double.
+     * If parsing fails, the default value is returned. This method is thread-safe
+     * and performs no network operations.
+     *
+     * @param flagKey The name of the flag to query. Cannot be null.
+     * @param defaultValue The value to return if the flag cannot be retrieved or parsed.
+     * @return The numeric value of the flag as a Number, or the default value if unavailable or unparseable.
+     */
+    override fun resolveNumberValue(flagKey: String, defaultValue: Number): Number {
+        val precomputedData = flagsRepository.getPrecomputedFlag(flagKey)
+        return precomputedData?.variationValue?.toDoubleOrNull() ?: defaultValue
+    }
+
+    /**
+     * Resolves a structured flag value from the local repository.
+     *
+     * This method attempts to parse the stored string value as a JSON object.
+     * If parsing fails, an error is logged and the default value is returned.
+     * This method is thread-safe and performs no network operations.
+     *
+     * @param flagKey The name of the flag to query. Cannot be null.
+     * @param defaultValue The value to return if the flag cannot be retrieved or parsed.
+     * @return The JSON object value of the flag, or the default value if unavailable or unparseable.
+     */
+    override fun resolveStructureValue(flagKey: String, defaultValue: JSONObject): JSONObject {
+        val precomputedData = flagsRepository.getPrecomputedFlag(flagKey)
+        return precomputedData?.variationValue?.let {
+            try {
+                JSONObject(it)
+            } catch (e: JSONException) {
+                featureSdkCore.internalLogger.log(
+                    level = InternalLogger.Level.ERROR,
+                    target = InternalLogger.Target.MAINTAINER,
```

Comment:
can it be useful for the user as well? they may discover that they have a malformed JSON stored by seeing the log